image/image.go

```diff
@@ -813,3 +813,25 @@ func (img *Image) Bin() ([]byte, error) {
 
 	return b.Bytes(), nil
 }
+
+// HasEncryptionPayload indicates whether an image's contains a HW encryption payload.
+func (img *Image) HasEncryptionPayload() bool {
+	enc := false
+	res, _ := img.FindAllUniqueTlv(IMAGE_TLV_SECRET_ID_LEGACY)
+	if res != nil {
+		nonce, _ := img.FindAllUniqueTlv(IMAGE_TLV_AES_NONCE_LEGACY)
+		if nonce != nil {
+			enc = true
+		}
+
+	}
```

Comment:
The block below only needs to run if `!enc`, right?

Reply:
there are two separate tlv indicators that could point to a hw enc. the legacy secret id or the current secret id, we need to check for both.

Reply:
both if blocks will always be run but once enc is set to true it cannot be unset

Reply:
Sure, but my point was that if you already know there is a legacy secret/nonce, you don't need to also look for a (non-legacy) secret/nonce, or am I missing something?

Reply:
no. I've optimized the solution using your feedback